src/midi/fluid_seq.cpp

```diff
@@ -39,44 +57,251 @@
 
 #define FLUID_SEQUENCER_EVENTS_MAX	1000
 
+/* Private data for clients */
+typedef struct _fluid_sequencer_client_t
+{
+    fluid_seq_id_t id;
+    std::string name;
+    fluid_event_callback_t callback;
+    void *data;
+} fluid_sequencer_client_t;
+
 /* Private data for SEQUENCER */
 struct _fluid_sequencer_t
 {
     // A backup of currentMs when we have received the last scale change
-    unsigned int startMs;
+    unsigned int startMs = 0;
 
     // The number of milliseconds passed since we have started the sequencer,
     // as indicated by the synth's sample timer
-    fluid_atomic_int_t currentMs;
+    std::atomic<int> currentMs{0};
 
     // A backup of cur_ticks when we have received the last scale change
-    unsigned int start_ticks;
+    unsigned int start_ticks = 0;
 
     // The tick count which we've used for the most recent event dispatching
-    unsigned int cur_ticks;
+    unsigned int cur_ticks = 0;
 
-    int useSystemTimer;
+    int useSystemTimer = 0;
 
     // The current time scale in ticks per second.
     // If you think of MIDI, this is equivalent to: (PPQN / 1000) / (USPQN / 1000)
-    double scale;
+    double scale = 1000;
 
-    fluid_list_t *clients;
-    fluid_seq_id_t clientsID;
+    std::vector<std::unique_ptr<fluid_sequencer_client_t>> clients;
+    fluid_seq_id_t clientsID = 0;
 
     // Pointer to the C++ event queue
-    void *queue;
-    fluid_rec_mutex_t mutex;
-};
+    void *queue = nullptr;
+    std::recursive_mutex mutex;
+    
+    _fluid_sequencer_t(bool use_system_timer)
+    {
+        if(use_system_timer)
+        {
+            FLUID_LOG(FLUID_WARN, "sequencer: Usage of the system timer has been deprecated!");
+        }
 
-/* Private data for clients */
-typedef struct _fluid_sequencer_client_t
-{
-    fluid_seq_id_t id;
-    char *name;
-    fluid_event_callback_t callback;
-    void *data;
-} fluid_sequencer_client_t;
+        this->useSystemTimer = use_system_timer ? 1 : 0;
+        this->startMs = this->useSystemTimer ? fluid_curtime() : 0;
+
+        this->queue = new_fluid_seq_queue(FLUID_SEQUENCER_EVENTS_MAX);
+        if(this->queue == NULL)
+        {
+            FLUID_LOG(FLUID_PANIC, "sequencer: Out of memory\n");
+            throw std::bad_alloc();
+        }
+    }
+    
+    ~_fluid_sequencer_t()
+    {
+        /* cleanup clients */
+        while(!this->clients.empty())
+        {
+            this->unregister_client(this->clients.back()->id);
+        }
+        delete_fluid_seq_queue(this->queue);
+    }
+    
+    fluid_seq_id_t register_client(const char *name, fluid_event_callback_t callback, void *data)
+    {
+        std::unique_ptr<fluid_sequencer_client_t> client(new fluid_sequencer_client_t);
```

Comment:
you can do auto client = std::make_unique<fluid_sequencer_client_t>() which is the more idiomatic way

Reply:
std::make_unique is C++14, which is not applicable here.